pkg/controller/seed-controller-manager/kubernetes/resources.go

```diff
@@ -230,6 +230,8 @@ func (r *Reconciler) getClusterTemplateData(ctx context.Context, cluster *kuberm
 		WithDnatControllerImage(r.dnatControllerImage).
 		WithMachineControllerImageTag(r.machineControllerImageTag).
 		WithMachineControllerImageRepository(r.machineControllerImageRepository).
+		WithOperatingSystemManagerImageTag(config).
+		WithOperatingSystemManagerImageRepository(config).
```

Comment:
Do we even need those 2 functions? Is there ever a case where we want to override the images so they do not match what is in the KubermaticConfiguration?

Reply:
Yeah, those are the setter functions for the TemplateData field, which is used to change the values of the osm specs. Otherwise I need to pass the KubermaticConfigurations variable everywhere, so instead I am injecting those overrides via the template data, which is already present in the function call.

Reply:
But we're already setting the config with `WithKubermaticConfiguration()`. If the values we set never diverge from the KubermaticConfiguration, then you can simply implement

```
func (d *TemplateData) OperatingSystemManagerImageTag() string {
	return d.config.Spec.OperatingSystemManager.ImageTag
}

func (d *TemplateData) OperatingSystemManagerImageRepository() string {
	return d.config.Spec.OperatingSystemManager.ImageRepository		
}
```

and don't need 2 extra setters.